momepy/utils.py

```diff
@@ -222,9 +222,11 @@ def _points_to_gdf(net, spatial_weights):
     Helper for nx_to_gdf.
     """
     node_xy, node_data = zip(*net.nodes(data=True))
-    gdf_nodes = gpd.GeoDataFrame(
-        list(node_data), geometry=[Point(i, j) for i, j in node_xy]
-    )
+    if len(node_xy[0]) == 2:
```

Comment:
Why not use unpacking like `geometry=[Point(*p) for p in node]` ? and maybe a check that the length is at most 3 due to the limitation of the shapely Point object

Reply:
That's way more elegant than my solution. In that case check is not needed as shapely will raise error anyway. I don't know what else other than error we could do with 4 values here, so it is fine. I'll change this in an upcoming PR adding more compatibility with `osmnx`, it is the same piece of code. Thanks!

Reply:
@martinfleis  Do you have a util for getting the the nearest `Point` on the line? Basically the orthogonal projection? I want to be able to snap points to it's nearest point on a street network and have them added as a nodes in a generated network. I looked  shortly at `pysal spaghetti` but its very slow :/

Reply:
@LeonardAukea not exactly. I have a function finding nearest line to a point, but it does not project it - https://github.com/martinfleis/momepy/blob/42fc80fc66c1a77035272e30eab04be075495867/momepy/elements.py#L643

Once you have the nearest line you can try to get the point using shapely (https://stackoverflow.com/a/24440122/9978945), but I am not sure if it will be faster than spaghetti. I want to do it at one point as it allows measuring reach from points along the network, but it likely won't happen soon.